### PR TITLE
[FLINK-11858][table-runtime-blink] Introduce block compression to batch table runtime

### PR DESCRIPTION
**Pull request by KurtYoung**
## What is the purpose of the change

Introduce BlockCompressor and BlockDecompressor to batch table runtime for future usage.

## Brief change log

- Add BlockCompressor interface
- Add BlockDecompressor interface
- Add LZ4 implementation based on the interfaces

## Verifying this change

Added unit tests

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes, lz4-java is introduced which is Apache license)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (yes)
  - If yes, how is the feature documented? (JavaDocs)

### CODE REVIEW

**Comment by KurtYoung** [comment]
cc @JingsongLi

File: flink-table/flink-table-runtime-blink/src/test/java/org/apache/flink/table/runtime/compression/BlockCompressionTest.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+import sun.nio.ch.DirectBuffer;
+
+import java.nio.ByteBuffer;
+
+import static org.junit.Assert.assertEquals;
+
+public class BlockCompressionTest {
+
+	@Test
+	public void testLz4() {
```
Comment (JingsongLi) [inline flink-table/flink-table-runtime-blink/src/test/java/org/apache/flink/table/runtime/compression/BlockCompressionTest.java]: test DataCorruptionException & InsufficientBufferException?

**Comment by KurtYoung** [comment]
Hi @JingsongLi , thanks for the review, i have addressed your comment.

**Comment by JingsongLi** [comment]
+1 LGTM

File: flink-table/flink-table-runtime-blink/pom.xml [hunk 2]
```diff
@@ -61,6 +61,13 @@ under the License.
 			<version>${janino.version}</version>
 		</dependency>
 
+		<!-- Lz4 compression library -->
```
Comment (twalthr) [inline flink-table/flink-table-runtime-blink/pom.xml]: @KurtYoung please also update the LICENSE file. Here is a rough description: https://cwiki.apache.org/confluence/display/FLINK/Licensing

Comment (KurtYoung) [reply flink-table/flink-table-runtime-blink/pom.xml]: ok, will do

**Pull request closed without merging**